File: src/main/java/no/nav/fo/veilarbregistrering/arbeidssoker/EndretFormidlingsgruppeCommand.java
package no.nav.fo.veilarbregistrering.arbeidssoker;

import no.nav.fo.veilarbregistrering.bruker.Foedselsnummer;

import java.time.LocalDateTime;
import java.util.Optional;

public interface EndretFormidlingsgruppeCommand {

    Optional<Foedselsnummer> getFoedselsnummer();

    String getPersonId();

    String getPersonIdStatus();

    Operation getOperation();

    Formidlingsgruppe getFormidlingsgruppe();

    LocalDateTime getFormidlingsgruppeEndret();

    Optional<Formidlingsgruppe> getForrigeFormidlingsgruppe();

    Optional<LocalDateTime> getForrigeFormidlingsgruppeEndret();

}

File: src/main/kotlin/no/nav/fo/veilarbregistrering/arbeidssoker/formidlingsgruppe/EndretFormidlingsgruppeCommand.kt
package no.nav.fo.veilarbregistrering.arbeidssoker.formidlingsgruppe

import no.nav.fo.veilarbregistrering.arbeidssoker.Operation
import no.nav.fo.veilarbregistrering.bruker.Foedselsnummer
import java.time.LocalDateTime

interface EndretFormidlingsgruppeCommand {
    val foedselsnummer: Foedselsnummer?
    val personId: String
    val personIdStatus: String?
    val operation: Operation
    val formidlingsgruppe: Formidlingsgruppe
    val formidlingsgruppeEndret: LocalDateTime
    val forrigeFormidlingsgruppe: Formidlingsgruppe?
    val forrigeFormidlingsgruppeEndret: LocalDateTime?
}